master/internal/api_experiment.go

```diff
@@ -1399,6 +1399,275 @@ func (a *apiServer) createUnmanagedExperimentTx(
+			fmt.Errorf("parsing override config: %w", err).Error())
+	}
+
+	if providedConfig.RawProject != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "'project' in override config "+
+			"cannot be specified, use `det experiment move` first if you want to change the project")
+	}
+	if providedConfig.RawWorkspace != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "'workspace' in override config "+
+			"cannot be specified, use `det experiment move` first if you want to change the workspace")
+	}
+
+	activeConfig, err := a.m.db.ActiveExperimentConfig(expID)
+	if err != nil {
+		return nil, fmt.Errorf("loading active config for experiment %d: %w", expID, err)
+	}
+	if name := activeConfig.Searcher().AsLegacy().Name; name != "single" { //nolint: goconst
+		return nil, status.Errorf(codes.InvalidArgument,
+			fmt.Sprintf(
+				"cannot continue a '%s' searcher experiment, must be a single searcher experiment",
+				name))
+	}
+
+	mergedConfig := schemas.Merge(providedConfig, activeConfig)
+	if name := mergedConfig.Searcher().AsLegacy().Name; name != "single" { //nolint: goconst
+		return nil, status.Errorf(codes.InvalidArgument,
+			fmt.Sprintf("override config must have single searcher type got '%s' instead", name))
+	}
+
+	bytes, err := mergedConfig.Value()
+	if err != nil {
+		return nil, fmt.Errorf("getting value of merged config: %w", err)
+	}
+
+	return bytes.([]byte), nil
+}
+
+func (a *apiServer) errorIfSearcherIsDone(trialID int, e expconf.ExperimentConfig) error {
+	latestCheckpoint, err := a.m.db.LatestCheckpointForTrial(trialID)
+	if err != nil {
+		return fmt.Errorf("getting latest checkpoint for trialID %d: %w", trialID, err)
+	}
+	if latestCheckpoint == nil || latestCheckpoint.StepsCompleted == 0 {
+		return nil // Can't be done if we don't have a checkpoint.
+	}
+
+	if e.Searcher().RawSingleConfig == nil {
+		return fmt.Errorf("something went wrong searcher single config is nil")
+	}
+
+	var inBatches int
+	units := int(e.Searcher().RawSingleConfig.MaxLength().Units)
+	unit := e.Searcher().Unit()
+	switch unit {
+	case expconf.Unitless, expconf.Unspecified:
+		return nil // Unitless makes us have no idea if the searcher is done.
+	case expconf.Records:
+		hyper, ok := e.Hyperparameters()["global_batch_size"]
+		if !ok {
+			return fmt.Errorf("searcher in records but did not specify global_batch_size")
+		}
+		if hyper.RawConstHyperparameter == nil {
+			return fmt.Errorf("global_batch_size not constant parameter")
+		}
+		globalBatchSize, ok := hyper.RawConstHyperparameter.RawVal.(int)
+		if !ok {
+			return fmt.Errorf("global_batch_size is not an integer got %v",
+				hyper.RawConstHyperparameter.RawVal)
+		}
+
+		inBatches = units / globalBatchSize
+		if inBatches < 1 {
+			inBatches = 1
+		}
+
+	case expconf.Batches:
+		inBatches = units
+
+	case expconf.Epochs:
+		hyper, ok := e.Hyperparameters()["global_batch_size"]
+		if !ok {
+			return fmt.Errorf("searcher in epochs but did not specify global_batch_size")
```

Comment:
`global_batch_size` is no longer required for PyTorch trials since `0.21.0`

Reply:
this code is getting deleted in favor for fixing the max length issue